### DISCUSSION TITLE
Running OSMT Quickstart from Windows

### DISCUSSION THREAD

**Discussion by JohnKallies** [wgu-opensource/osmt#131] [Helps to know] [answered]
To start a demo OSMT using the Quickstart, you will need to have [git](https://git-scm.com/downloads) and [Docker](https://docs.docker.com/get-docker/) installed, with at least 6GB allocated to Docker. More memory will be better.

* Installing Docker on Windows is a manageable task, but it is not a simple double-click installation. Starting a demo OSMT via Quickstart is simpler in MacOS or in Linux. If you are a Windows user, you can install Docker with [these instructions](https://docs.docker.com/desktop/windows/install/).

## Windows and Quickstart
Download the OSMT source code repo with this command:

```git clone https://github.com/wgu-opensource/osmt.git```

### Using the OSMT development utility (osmt_cli.sh)
The OSMT source code includes a utility named osmt_cli.sh; this simplifies setting up an OSMT environment and doing routine tasks. You can run via a terminal window, from the OSMT git repo's root directory. osmt_cli.sh uses BASH, and works on MacOS and Linux. Using Windows, you may need to use a program like WSL2, "Git Bash" or CygWin. Run this command to see the help output:

```./osmt_cli.sh -h```

If you are using Windows, and having trouble running this help command, please reach out here in the Discussion area.

**Comment by bredd** [comment, marked as answer]
For running "osmt_cli.sh" on Windows, Git Bash works best for me.

**Reply by JohnKallies** [reply]
@bredd Thank you for the feedback.

I found on my spare Windows 10 desktop tucked away in my basement, with a fresh Docker Desktop installed, I needed to change my env files, changing the underscores in the service URLs to hyphens to match what I go from `docker ps`. Did you have to do that as well?
